From 23002defd462845db376425a7b7d975f3deba83d Mon Sep 17 00:00:00 2001
From: Neal Frager <neal.frager@amd.com>
Date: Mon, 24 Apr 2023 12:53:25 +0100
Subject: [PATCH v1 1/1] pmufw: misc/Makefile: specify sequential Makefiles

The BSP_SEQUENTIAL_MAKEFILES variable is not properly assigned and exported
from copy_bsp.sh.

Because of this, no library is built sequentially even if it was desired to
build them sequentially by assigning to BSP_SEQUENTIAL_MAKEFILES. All the
libraries are built in parallel.

This patch resolves this issue, so that libraries that must be built
sequentially are indeed built sequentially.

Signed-off-by: Neal Frager <neal.frager@amd.com>
---
 lib/sw_apps/zynqmp_pmufw/misc/Makefile    | 10 +++++++++-
 lib/sw_apps/zynqmp_pmufw/misc/copy_bsp.sh |  7 -------
 lib/sw_apps/zynqmp_pmufw/src/Makefile     |  2 ++
 3 files changed, 11 insertions(+), 8 deletions(-)

diff --git a/lib/sw_apps/zynqmp_pmufw/misc/Makefile b/lib/sw_apps/zynqmp_pmufw/misc/Makefile
index a773498512..fe2d23c965 100644
--- a/lib/sw_apps/zynqmp_pmufw/misc/Makefile
+++ b/lib/sw_apps/zynqmp_pmufw/misc/Makefile
@@ -7,6 +7,14 @@ PROCESSOR = psu_pmu_0
 LIBRARIES = ${PROCESSOR}/lib/libxil.a
 BSP_MAKEFILES := $(wildcard $(PROCESSOR)/libsrc/*/src/Makefile)
 SUBDIRS := $(patsubst %/Makefile, %, $(BSP_MAKEFILES))
+DRIVERS_LIST=../drivers.txt
+SEQUENTIAL_MAKEFILES := $(shell cat ${DRIVERS_LIST})
+BSP_SEQUENTIAL_MAKEFILES = $(patsubst %, ${PROCESSOR}/libsrc/%/src/Makefile, $(SEQUENTIAL_MAKEFILES))
+BSP_SEQUENTIAL_MAKEFILES += ${PROCESSOR}/libsrc/xilskey/src/Makefile
+BSP_SEQUENTIAL_MAKEFILES += ${PROCESSOR}/libsrc/xilfpga/src/Makefile
+BSP_SEQUENTIAL_MAKEFILES += ${PROCESSOR}/libsrc/xilsecure/src/Makefile
+BSP_SEQUENTIAL_MAKEFILES += ${PROCESSOR}/libsrc/dppsu/src/Makefile
+BSP_SEQUENTIAL_MAKEFILES += ${PROCESSOR}/libsrc/dpdma/src/Makefile
 BSP_PARALLEL_MAKEFILES := $(filter-out $(BSP_SEQUENTIAL_MAKEFILES),$(BSP_MAKEFILES))
 SEQ_SUBDIRS := $(patsubst %/Makefile, %, $(BSP_SEQUENTIAL_MAKEFILES))
 PAR_SUBDIRS := $(patsubst %/Makefile, %, $(BSP_PARALLEL_MAKEFILES))
@@ -16,7 +24,7 @@ ifneq (,$(findstring win,$(RDI_PLATFORM)))
 endif
 
 all:
-	$(MAKE) --no-print-directory seq_libs
+	$(MAKE) -j1 --no-print-directory seq_libs
 	$(MAKE) -j --no-print-directory par_libs
 	$(MAKE) --no-print-directory archive
 	@echo 'Finished building libraries'
diff --git a/lib/sw_apps/zynqmp_pmufw/misc/copy_bsp.sh b/lib/sw_apps/zynqmp_pmufw/misc/copy_bsp.sh
index 197f7af844..ac8dd8249e 100755
--- a/lib/sw_apps/zynqmp_pmufw/misc/copy_bsp.sh
+++ b/lib/sw_apps/zynqmp_pmufw/misc/copy_bsp.sh
@@ -29,8 +29,6 @@ STANDALONE_DIR=$EMBEDDED_SW_DIR/lib/bsp/standalone/src
 # libraries dir
 SERVICES_DIR=$EMBEDDED_SW_DIR/lib/sw_services
 
-BSP_SEQUENTIAL_MAKEFILES=
-
 # creation of BSP folders required
 if [ -d $BSP_DIR ]; then
 	echo "BSP directory already exists"
@@ -55,14 +53,12 @@ cp -r $SERVICES_DIR/xilfpga/src/interface/zynqmp/xilfpga_pcap.c $BSP_DIR/libsrc/
 cp -r $SERVICES_DIR/xilfpga/src/*.h $BSP_DIR/include/
 cp -r $SERVICES_DIR/xilfpga/src/interface/zynqmp/*.h $BSP_DIR/include/
 rm -r $BSP_DIR/libsrc/xilfpga/src/interface/
-BSP_SEQUENTIAL_MAKEFILES="$BSP_SEQUENTIAL_MAKEFILES $BSP_DIR/libsrc/xilfpga/src/Makefile"
 mkdir -p $BSP_DIR/libsrc/xilsecure/src/
 cp -r $SERVICES_DIR/xilsecure/src/Makefile $BSP_DIR/libsrc/xilsecure/src/
 cp -r $SERVICES_DIR/xilsecure/src/common/all/* $BSP_DIR/libsrc/xilsecure/src/
 cp -r $SERVICES_DIR/xilsecure/src/zynqmp/* $BSP_DIR/libsrc/xilsecure/src/
 cp -r $SERVICES_DIR/xilsecure/src/common/all/*.h $BSP_DIR/include/
 cp -r $SERVICES_DIR/xilsecure/src/zynqmp/*.h $BSP_DIR/include/
-BSP_SEQUENTIAL_MAKEFILES="$BSP_SEQUENTIAL_MAKEFILES $BSP_DIR/libsrc/xilsecure/src/Makefile"
 cp -r $SERVICES_DIR/xilskey/ $BSP_DIR/libsrc/
 
 # remove the xilskey library files which are not required for PMU
@@ -84,7 +80,6 @@ rm -r $BSP_DIR/libsrc/xilskey/src/include/xilskey_bbram.h
 # copy the xilskey library header files to include directory
 cp -r $BSP_DIR/libsrc/xilskey/src/*.h $BSP_DIR/include/
 cp -r $BSP_DIR/libsrc/xilskey/src/include/*.h $BSP_DIR/include/
-BSP_SEQUENTIAL_MAKEFILES="$BSP_SEQUENTIAL_MAKEFILES $BSP_DIR/libsrc/xilskey/src/Makefile"
 
 # copy bsp standalone code
 cp -r $STANDALONE_DIR/common/*  $BSP_DIR/libsrc/standalone/src/
@@ -113,7 +108,6 @@ do
 	if [ $line != "avbuf" ] && [ $line != "video_common" ]; then
 		cp $WORKING_DIR/x"$line"_g.c $BSP_DIR/libsrc/$line/src/
 	fi
-	BSP_SEQUENTIAL_MAKEFILES="$BSP_SEQUENTIAL_MAKEFILES $BSP_DIR/libsrc/$line/src/Makefile"
 
 done < $DRIVERS_LIST
 
@@ -136,4 +130,3 @@ cp $STANDALONE_DIR/profile/*.h  $BSP_DIR/include/
 
 # no inbyte and outbyte present in standalone
 cp $WORKING_DIR/inbyte.c $WORKING_DIR/outbyte.c  $BSP_DIR/libsrc/standalone/src/
-export BSP_SEQUENTIAL_MAKEFILES
diff --git a/lib/sw_apps/zynqmp_pmufw/src/Makefile b/lib/sw_apps/zynqmp_pmufw/src/Makefile
index 1750c0a329..8747db5cdf 100644
--- a/lib/sw_apps/zynqmp_pmufw/src/Makefile
+++ b/lib/sw_apps/zynqmp_pmufw/src/Makefile
@@ -27,6 +27,8 @@ all: $(EXEC)
 $(EXEC): $(LIBS) $(OBJS) $(INCLUDES)
 	$(CC) -o $@ $(OBJS) $(CC_FLAGS) $(CFLAGS) $(LN_FLAGS) $(LIBPATH) $(LSCRIPT)
 
+$(OBJS): $(LIBS)
+
 $(LIBS):
 	echo "Copying BSP files"
 	../misc/copy_bsp.sh
-- 
2.17.1

